packages/element-lib/src/sidetree-v2/protocol/resolve.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable arrow-body-style */
+const jsonpatch = require('fast-json-patch');
+const { payloadToHash } = require('../func');
+
+const reducer = (state = {}, operation) => {
+  const type = operation.decodedOperation.header.operation;
+  switch (type) {
+    case 'create':
+      return {
+        ...operation.decodedOperationPayload,
+        id: `did:elem:${payloadToHash(operation.decodedOperationPayload)}`,
+      };
+    case 'update':
+      return operation.decodedOperationPayload.patch.reduce(jsonpatch.applyReducer, state);
+    case 'recover':
```

Comment:
I know that from a compatibility perspective, sidetree-core may not have this functionality implemented, but this makes me a little sad.